### PR TITLE
Clean up train script and copy model creation and train functions to MIMIC demo notebook

### PR DESCRIPTION
**Pull request by cx1111**
* Create standalone train_model function to hold logic of train.py script. Allows calling as function or script.

* Rename some of the train.py command line flags for clarity. Reorder flags/params to be consistent.

* Add a flag for the saved model file name

* Replace "tune" with "validate" in function params, flags, and call sites

* MIMIC notebook: add copy of create_model and train_model functions to more easily demonstrate code

* CXR_Foundation notebook: replace the Pypi install with a git clone and pip install to prevent drift

### CODE REVIEW

**Comment by cx1111** [comment]
Explanation:
- I replaced the word "tune" with "validate" everywhere since the only TF param that references the concept is `model.fit(validation_data=xxx)`, and because the MIMIC JPG split labels were also named "validate". Please revert and LMK if I'm using the ML term incorrectly.

- I figured that we can avoid using the PyPi package until this package stabilizes a bit more after the changes that'll be made for the upcoming datathons. Or perhaps it'd be less work for the package maintainers in general (ie. just you) to just make users install from GH.


Review/testing instructions:
- The code is from my fork, on branch `mimic-demo`
- View notebook in Colab at: https://colab.research.google.com/github/cx1111/imaging-research/blob/mimic-demo/cxr-foundation/MIMIC_Embeddings_Demo.ipynb
- In the notebooks, change the git clone line to: `!git clone -b mimic-demo https://github.com/cx1111/imaging-research.git` while testing. They should keep referring to the main repo/branch when merged.
- Note: I just learned that VSCode w/ Python plugins generates a pretty nice visual for the notebook diff from the Source Control tab

**TODOs BEFORE merging** (if everything else looks correct):
- Format the python files with Google py format (I don't have access to it right now)


Fuure to dos:
- Increase the repo version
- The `heads` param of create_model is rather confusing. May be good to refactor.
- Is there a way let users see the absl flag docstrings when they call help on the module?

**Comment by cx1111** [comment]
I'm out of time. Final thoughts:
- I ran the MIMIC_demo notebook in colab just once, and it worked. <insert meme about something being wrong>.
- I didn't run the other CXR_Foundation notebook, so it will need checking. But I also just realized that changing this notebook doesn't have to go in the same PR, since it installs the fixed cxr-foundation package from PyPi, so the changes here shouldn't break it.

So feel free to change what's needed, revert the notebook, reword the commit, force push onto this branch, etc. This is the state I'm going to leave this PR in. Thanks!
